main.c

```diff
@@ -143,6 +143,8 @@ int hydro_settings(struct corsair_device_scan scanned_device, struct option_pars
 			uint8_t v2 = (temperature&0xFF);
 			msg_debug("DEBUG: %02X %02X\n", v1, v2);
 			celsius = (double)v1 + ((double)v2/10);
+		} else if (dev->driver == &corsairlink_driver_hid) {
+			celsius = (double)temperature / 256;
```

Comment:
The temperature is already in Celsius, so making a conversion to Celsius isn't needed. Also, why divide by 256?

Reply:
Looking at my legacy code 256 is correct.